### ISSUE TITLE
Addition of code for time series data

### ISSUE THREAD

**Issue by kbelisar** [ccodwg/Covid19CanadaDataProcess#49]
## Addition of Time Series Data

The processing of time series data for the four main Data Domains (Cases, Mortality, Hospitalizations, and ICU) is required for both inactive and active data sets. The time series data required are listed below using the following format:

PROVINCE / Data Level (Prov = provincial or HR = health region) / Data Domain / UUID

### Active Data:

- [x] CAN / NA / Cases / 558ccc64-799f-4329-95a4-f944dac21eb1
- [x] AB / Prov / Cases / 3ced816d-8524-4875-bd69-61fb5603b596
- [x] AB / Prov / Mortality / 3ced816d-8524-4875-bd69-61fb5603b596
- [x] AB / Prov / Hospitalizations / 3ced816d-8524-4875-bd69-61fb5603b596
- [x] AB / Prov / ICU / 3ced816d-8524-4875-bd69-61fb5603b596
- [x] ~~#50~~
- [x] ~~BC / HR / Cases / 4f9dc8b7-7b42-450e-a741-a0f6a621d2af~~ (historical)
- [x] NB / Prov / Cases / 40d93d2b-5c85-4592-b844-fb78b5ae5cec
- [x] NB / Prov / Mortality / 40d93d2b-5c85-4592-b844-fb78b5ae5cec
- [x] NB / Prov / Hospitalizations / 40d93d2b-5c85-4592-b844-fb78b5ae5cec
- [x] NB / Prov / ICU / 40d93d2b-5c85-4592-b844-fb78b5ae5cec
- [x] NL / Prov / Cases /13a35524-89bf-46a2-afc5-d88ef81bfa82
- [x] NL / Prov / Mortality /13a35524-89bf-46a2-afc5-d88ef81bfa82
- [x] NL / Prov / ICU /13a35524-89bf-46a2-afc5-d88ef81bfa82
- [x] QC / Prov / Mortality / d7f9cb92-a83a-4cd3-b955-762410b2cf5c
- [x] QC / Prov / Cases /  d546a7f6-27ac-4305-bad6-b04435211e79
- [x]  QC / Prov / Mortality /  d546a7f6-27ac-4305-bad6-b04435211e79
- [x] QC / Prov / Hospitalizations / b78d46c8-9a56-4b75-94c5-4ace36e014f5
- [x] QC / Prov / ICU / b78d46c8-9a56-4b75-94c5-4ace36e014f5
- [x] QC / HR / Hospitalizations / 3b93b663-4b3f-43b4-a23d-cbf6d149d2c5
- [x] QC / HR / ICU / 3b93b663-4b3f-43b4-a23d-cbf6d149d2c5
- [x] ~~QC / Prov / Cases / 5c1ef873-2538-4f99-b8cf-102728c63fea~~ (historical)
- [x] ~~QC / Prov / Hospitalizations / bf806684-60ff-4f48-b34b-e292487586c7~~ (historical)
- [x] YT / Prov / Cases / 7698e84c-f43f-4a83-828a-1d26e21467b9
- [x] YT / Prov / Cases / af6ea226-0325-4800-bac2-9239dd507b49

**Comment by jeanpaulrsoucy** [comment]
Two notes.

We'll eventually have to discuss how to incorporate extra data (like cases by age groups, vaccination status, etc.) into the output datasets. We can probably figure that out later, but should probably make a note of which datasets this affects. I'm thinking something like a `group` column? Thoughts?

Second, to load inactive datasets, you have to use `dl_archive` rather than `dl_dataset`. For example:

```
ds <- Covid19CanadaData::dl_archive("5c1ef873-2538-4f99-b8cf-102728c63fea")
```

**Comment by kbelisar** [comment]
I'd lean towards having separate columns depending on the grouping type to make the data more accessible to users not familiar with that type of data format. It may also prove easier for documentation purposes when definitions differ. That is, by keeping grouping as distinct columns, a systematic approach to variable naming can be deployed.

As for the inactive datasets, thanks for the heads up!

**Comment by jeanpaulrsoucy** [comment]
> I'd lean towards having separate columns depending on the grouping type to make the data more accessible to users not familiar with that type of data format. It may also prove easier for documentation purposes when definitions differ. That is, by keeping grouping as distinct columns, a systematic approach to variable naming can be deployed.
> 
> As for the inactive datasets, thanks for the heads up!

I see your point, but I would prefer having common column names for all outputs. What about `group_name` and `group_value`. For example, “age group” and “0-19”. Could be NA and NA for datasets without groups.

**Comment by kbelisar** [comment]
> I see your point, but I would prefer having common column names for all outputs. What about `group_name` and `group_value`? For example, “age group” and “0-19”. Could be NA and NA for datasets without groups.

This is an adequate work-around and allows for the necessary detail to reside within the data set itself. I don't see any issues with this!

**Comment by benkcwong** [comment]
Having a "grouped" column works for me. Maybe we can have it formatted so that someone can split the text into separate groups if they wanted?

**Comment by jeanpaulrsoucy** [comment]
Either way, I'll need to add support for group-related columns in the `helper` functions, otherwise the columns get get tossed/throw errors. This also means adding downstream support for extra columns in `Covid19CanadaETL` so that it doesn't accidentally mess up the data automation pipeline. I'll keep you appraised of my progress.

**Comment by kbelisar** [comment]
> Second, to load inactive datasets, you have to use `dl_archive` rather than `dl_dataset`. For example:
> 
> ```
> ds <- Covid19CanadaData::dl_archive("5c1ef873-2538-4f99-b8cf-102728c63fea")
> ```

An update: The previously "Inactive" data sets are still active, they just have a lag of ~3 days before the data are available. So fortunately this means that there were no changes to the frequency of data being available as initially thought and we can continue to collect daily numbers!

**Comment by jeanpaulrsoucy** [comment]
Some code clean-up & fixes in 46f144efbc48d20db83f086f32c342be48dc840b.
